File: src/main/java/be/superjoran/mint/services/BankAccountService.java
package be.superjoran.mint.services;

import be.superjoran.common.model.DomainObjectCrudService;
import be.superjoran.mint.domain.BankAccount;
import be.superjoran.mint.domain.Person;
import be.superjoran.mint.domain.searchresults.BankAccountCandidate;

import java.util.List;

public interface BankAccountService extends DomainObjectCrudService<BankAccount> {

    List<BankAccount> findAllByOwner(Person administrator);

    BankAccount createOrUpdate(BankAccountCandidate bankAccountCandidate);
}

File: src/main/java/be/superjoran/mint/services/BankAccountService.kt
package be.superjoran.mint.services

import be.superjoran.common.model.DomainObjectCrudService
import be.superjoran.mint.domain.BankAccount
import be.superjoran.mint.domain.Person
import be.superjoran.mint.domain.searchresults.BankAccountCandidate

interface BankAccountService : DomainObjectCrudService<BankAccount> {

    fun findAllByOwner(administrator: Person): List<BankAccount>

    fun create(bankAccountCandidate: BankAccountCandidate): BankAccount
}
